mpr-research-data.py

```diff
@@ -1,51 +1,220 @@
+                           'PRIVATE_KEY': None,
+                           'CLIENT_EMAIL': 'dbtogcphelper@mwrite-a835.iam.gserviceaccount.com',
+                           'CLIENT_ID': '101781732898615574177',
+                           'AUTH_URI': 'https://accounts.google.com/o/oauth2/auth',
+                           'TOKEN_URI': 'https://oauth2.googleapis.com/token',
+                           'AUTH_PROVIDER_X509_CERT_URL': 'https://www.googleapis.com/oauth2/v1/certs',
+                           'CLIENT_X509_CERT_URL': 'https://www.googleapis.com/robot/v1/metadata/x509/dbtogcphelper%40mwrite-a835.iam.gserviceaccount.com'}
+    gcpCredsDict = {}
+
+    allKeyPartsFound = True
+    for credPart in gcpCredsDefaultDict:
+        gcpCredsDict[credPart.lower()] = os.getenv(
+            'GCP_' + credPart, gcpCredsDefaultDict[credPart])
+
+        if not gcpCredsDict[credPart.lower()]:
+            logging.error(
+                f'Did not find .env variable for Service Account GCP key: {credPart}')
+            allKeyPartsFound = False
+
+    if not allKeyPartsFound:
+        sys.exit('Exiting.')
+
+    return gcpCredsDict
+
+
+def queryRetriever(queryName, queryModifier=False,
+                   queryFolder=defaultQueryFolder):
+    with open(os.path.join(queryFolder, queryName)) as queryFile:
+        queryLines = ''.join(queryFile.readlines())
+
+    if queryModifier:
+        queryLines = queryLines.format(queryModifier)
+
+    return queryLines
+
+
+def makeDBConnection(dbParams):
+
+    try:
+        connectString = f'mysql+pymysql://{dbParams["USER"]}:{dbParams["PASSWORD"]}@{dbParams["HOST"]}:{dbParams["PORT"]}/{dbParams["NAME"]}'
+        engine = sql.create_engine(connectString)
+        logging.info('DB connection established.')
+        return engine
+
+    except Exception as e:
+        logging.error(f'Error Message: {e}')
+        logging.error('Failed to establish DB connection.')
+        sys.exit('Exiting.')
+
+
+def makeGCPConnection(gcpParams):
+
+    try:
+        client = storage.Client.from_service_account_info(gcpParams)
+        logging.info('GCP connection established.')
+        return client
+
+    except Exception as e:
+        logging.error(f'Error Message: {e}')
+        logging.error('Failed to establish GCP connection.')
+        sys.exit('Exiting.')
+
+
+def courseQueryMaker(courseQueryTemplate, monthsModifier, engine):
+
+    try:
+        courseQuery = queryRetriever(courseQueryTemplate, monthsModifier)
+        with engine.connect() as connection:
+            courseDF = pd.read_sql(courseQuery, connection)
+        logging.info('Courses retrieved...')
+
+        return courseDF
+
+    except Exception as e:
+        logging.error(f'Error Message: {e}')
+        logging.error('Failed to retrieve Course List.')
+        sys.exit('Exiting.')
+
+
+def retrieveQueryMaker(retrieveQueryTemplate, courseModifier, engine):
+
+    try:
+        courseIDString = ',\n   '.join(map(str, courseModifier)) + '\n'
```

Comment:
I suggest naming variables a little more specifically, if they're known to always contain specific data.

Also, although the SQL query is nicely formatted, the parameter substitution doesn't need to be neat also.  Processing the substitution can be a little simpler.

```suggestion
def retrieveQueryMaker(retrieveQueryTemplate, courseIDs, engine):

    try:
        courseIDString = ','.join(map(str, courseIDs))
```